tests/contrib/fastapi/test_fastapi.py

```diff
@@ -30,9 +32,11 @@ def tracer():
 
     setattr(ddtrace, "tracer", tracer)
     fastapi_patch()
+    starlette_patch()
```

Comment:
hmm... we shouldn't need this should we?

Reply:
Without this we get no obfuscation of the path for example:
 `FAILED tests/contrib/fastapi/test_fastapi.py::test_read_item_success - AssertionError: assert 'GET /items/foo' == 'GET /items/{item_id}'`
 
 Only by patching starlette as well do we get the correct paths. I originally tried patching the FastAPI additional handlers e.g https://github.com/tiangolo/fastapi/blob/2b54432a9c78730b0dd69fca8f15d932d5a31d6b/fastapi/routing.py#L307 but it doesn't cover all cases and ends up with double wrapping if we patch starlette as well. 
 
 Without patching starlette, I don't understand how the handlers of Fastapi will be patched.

Reply:
hmm, oh, this was probably working before because FastAPI had a `span_modifier` defined, so it didn't require Starlette to be patched in order to work.

Now with this change... are we now *requiring* Starlette to be patched in order to use FastAPI?

Should we replicate some of the patching in Starlette into FastAPI?

Reply:
Going to look more into why instrumenting the fastapi handlers isn't sufficient. Going to confirm odd behavior and clarify it.

Reply:
It seems we just need to solve the double wrapping due to Starlette instrumentation wrapping `Mount.handle`, which Fastapi’s routing uses directly, and we end up wrapping again in Fastapi instrumentation: https://github.com/tiangolo/fastapi/blob/master/fastapi/routing.py#L49 .
Here we can see in the traceback that it’s wrapped twice:  https://a.cl.ly/4gur0BJm 

Options to solve:

1. Would the best solution here be to just not patch fastapi.routing.Mount.handle since they use exactly the same Mount.handle function as Starlette? Of course doing this would again make Fastapi instrumentation reliant on Starlette instrumentation. Which I think we want to avoid.

2. Or, we could check in both the fastapi and starlette instrumentations if the Mount.handle function is already wrapped, if so, don’t wrap it again?

Reply:
Fixed with checking method isn't already wrapped using: `isinstance(the_thing, ObjectProxy)`